frontend/src/components/object/ObjectTag.vue

```diff
@@ -2,32 +2,63 @@
 import { storeToRefs } from 'pinia';
 import { onMounted, ref, watch } from 'vue';
 
-import { Button } from '@/lib/primevue';
-import { useTagStore, useVersionStore } from '@/store';
+import ObjectMetadataTagForm from '@/components/object/ObjectMetadataTagForm.vue';
+import { Button, Dialog } from '@/lib/primevue';
+import { useObjectStore, useTagStore, useVersionStore } from '@/store';
 
 import type { Ref } from 'vue';
+import type { ObjectMetadataTagFormType } from '@/components/object/ObjectMetadataTagForm.vue';
 import type { Tagging } from '@/types';
 
 // Props
 type Props = {
+  editable?: boolean;
   objectId: string;
   versionId?: string;
 };
 
 const props = withDefaults(defineProps<Props>(), {
+  editable: true,
   versionId: undefined
 });
 
+// Emits
+const emit = defineEmits(['on-file-uploaded']);
+
 // Store
 const tagStore = useTagStore();
 const versionStore = useVersionStore();
 const { getTagging: tsGetTagging } = storeToRefs(tagStore);
 const { getTagging: vsGetTagging } = storeToRefs(versionStore);
 
 // State
+const metaVisible: Ref<boolean> = ref(false);
```

Comment:
you could follow a convention that works for both metadata and tag
~~showMetaModal~~ showModal
~~closeMetaModal~~ closeModal
~~submitMetaModal~~ submit
~~metaVisible~~ editing
~~metadataTagFormData~~ formData

Reply:
I agree with this - if there is capacity to abstract and decontextualize the component so that it can be reused in multiple contexts, we should position that way.